src/dcat-ap/dcat-dataset.test.ts

```diff
@@ -0,0 +1,122 @@
+import { cloneObject } from '@esri/hub-common';
+import { DcatDataset } from './dcat-dataset';
+
+import * as datasetFromApi from './test-helpers/mock-dataset.json';
+
+const siteUrl = 'https://foobar.hub.arcgis.com'
+const orgTitle = 'My Fun Org'
+const portalUrl = 'https://my-fun-org.maps.arcgis.com'
+
+describe('DcatDataset', () => {
+  it('Dataset props come from right places', function() {
+    const dataset = new DcatDataset(datasetFromApi.attributes, portalUrl, orgTitle, siteUrl)
+
+    expect(dataset.id).toBe('f4bcc1035b7d46cba95e977f4affb6be_0')
+    expect(dataset.url).toBe(
+      'https://servicesqa.arcgis.com/Xj56SBi2udA78cC9/arcgis/rest/services/Tahoe_Things/FeatureServer/0'
+    )
+    expect(dataset.landingPage).toBe(
+      'https://foobar.hub.arcgis.com/datasets/f4bcc1035b7d46cba95e977f4affb6be_0'
+    )
+    expect(dataset.title).toBe('Tahoe places of interest')
+    expect(dataset.description).toBe('Description. Here be Tahoe things. You can do a lot here. Here are some more words. And a few more.<div><br /></div><div>with more words</div><div><br /></div><div>adding a few more to test how long it takes for our jobs to execute.</div><div><br /></div><div>Tom was here!</div>')
+    expect(dataset.owner).toBe('thervey_qa_pre_a_hub')
+    expect(dataset.ownerUri).toBe(
+      'https://my-fun-org.maps.arcgis.com/sharing/rest/community/users/thervey_qa_pre_a_hub?f=json'
+    )
+    expect(dataset.language).toBe('ger')
+    expect(dataset.keyword).toEqual([
+      "some",
+      "keywords",
+      "from",
+      "metadata"
+    ])
+    expect(dataset.issuedDateTime).toBe('2021-04-19T13:30:24.055-04:00')
+    expect(dataset.orgTitle).toBe('My Fun Org')
+    // TODO - update when this is available through the API
```

Comment:
`org.portalProperties.links.contactUs.url` is currently unavailable through the API.